### ISSUE TITLE
Store Request and Group IDs in backend key value store

### ISSUE THREAD

**Issue by hacktobeer** [google/turbinia#1030]
We currently use key/value stores as backend. This stores the key as 'TurbiniaTask:[task_id]' with a JSON blob as value containing the details (including request id and group id). If you want to query on request or group ID we loop through all keys and parse the value to get the relevant task IDs.

We could store the RequestID with a value list of TaskIDs and the GroupID with a value list of Request IDs to make status fetching faster.

**Comment by jleaniz** [comment]
Looking into implemting this change.  I will use this issue to cover any related bugs for simplicity.

The ```to_json()``` method in the ```TurbiniaRequest``` class needs to handle invalid evidence objects. It expects the evidence attribute to be a list, not an Evidence object. The name of the attribute is a bit misleading here as well and should consider renaming it to ```evidence_list``` or similar.
```
>>> r = RawDisk(source_path='/evidence/artifact_disk.dd')
>>> t = TurbiniaRequest(evidence=r, requester='leaniz')
>>> t
<turbinia.message.TurbiniaRequest object at 0x7fe97a2a0310>
>>> t.to_json()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/workspaces/turbinia/turbinia/message.py", line 69, in to_json
    serializable['evidence'] = [x.serialize() for x in serializable['evidence']]
TypeError: 'RawDisk' object is not iterable
```

**Comment by jleaniz** [comment]
Fixed in #1489
